test/e2e/framework/job/wait.go

```diff
@@ -68,6 +69,17 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobReady waits for particular value of the Job .status.ready field
+func WaitForJobReady(ctx context.Context, c clientset.Interface, ns, jobName string, ready *int32) error {
+	return wait.PollUntilContextTimeout(ctx, framework.Poll, JobTimeout, false, func(ctx context.Context) (bool, error) {
```

Comment:
Turns out, we are moving away from using `wait.Poll` in e2e tests.
Use `gomega.Eventually`

https://www.kubernetes.dev/blog/2023/04/12/e2e-testing-best-practices-reloaded/#polling-and-timeouts

Example: https://github.com/kubernetes/kubernetes/pull/121017

Reply:
replaced on `gomega.Eventually` https://github.com/kubernetes/kubernetes/pull/121300/commits/ec887a2c8a8e2b28db5e850aded5468c341b8897